core/src/main/scala/kafka/server/metadata/MetadataBrokers.scala

```diff
@@ -37,7 +36,15 @@ object MetadataBroker {
         endPoint.name() ->
           new Node(record.brokerId, endPoint.host, endPoint.port, record.rack)
       }.toMap,
-      true)
+      fenced = true)
+  }
+
+  def apply(broker: Broker): MetadataBroker = {
+    new MetadataBroker(broker.id, broker.rack.orNull,
+      broker.endPoints.map { endpoint =>
+        endpoint.listenerName.value -> new Node(broker.id, endpoint.host, endpoint.port, broker.rack.orNull)
+      }.toMap,
+      fenced = false)
```

Comment:
I wanted to go ahead and conform to MetadataBroker for both implementations. One side-effect is we are exposing the fenced flag to ZK-based clusters. I've set it to false here since we don't have any notion of broker fencing in the ZK-based metadata. Is there any harm in this?

Reply:
as long as this is just used by the ZK code path there is no harm